SampleApps/SampleApps/SampleEnclave/my_exports.cpp

```diff
@@ -283,6 +295,205 @@ HRESULT VbsEnclave::Trusted::Implementation::RunTaskpoolExample(_In_ const std::
+        convertedConfig.cacheUsageCount = keyCredentialCacheConfiguration.cacheUsageCount;
+
+        debug_print(L"Created mutableKeyCredentialCacheConfig");
+
+        auto keyBytes = veil::vtl1::userboundkey::enclave_create_user_bound_key(
+            helloKeyName,
+            convertedConfig,
+            pinMessage,
+            windowId,
+            ENCLAVE_SEALING_IDENTITY_POLICY::ENCLAVE_IDENTITY_POLICY_SEAL_SAME_IMAGE);
+
+        debug_print(L"enclave_create_user_bound_key returned");
+
+        // Store the user-bound key bytes directly - do NOT try to create a symmetric key from them
+        securedEncryptionKeyBytes.assign(keyBytes.begin(), keyBytes.end());
+        
+        // Do NOT try to create a symmetric key here - user-bound keys must be loaded properly
+        // g_encryptionKey will be set in the load functions
+    }
+    catch (const std::exception& e)
+    {
+        debug_print(L"Exception caught in MyEnclaveCreateUserBoundKey: %hs", e.what());
+        return E_FAIL;
+    }
+
+    return S_OK;
+}
+
+//
+// Load user bound key and encrypt data
+//
+HRESULT VbsEnclave::Trusted::Implementation::MyEnclaveLoadUserBoundKeyAndEncryptData(
+    _In_ const std::wstring& helloKeyName,
+    _In_ const std::wstring& pinMessage,
+    _In_ const uintptr_t windowId,
+    _In_ const keyCredentialCacheConfig& keyCredentialCacheConfiguration,
+    _In_ const std::vector<std::uint8_t>& securedEncryptionKeyBytes,
+    _In_ const std::wstring& inputData,
+    _Out_ std::vector<std::uint8_t>& encryptedInputBytes,
+    _Out_ std::vector<std::uint8_t>& tag)
+{
+    using namespace veil::vtl1::vtl0_functions;
+
+    try
+    {
+        debug_print(L"Start MyEnclaveLoadUserBoundKeyAndEncryptData");
+
+        // Only load the user-bound key if it's not already loaded
+        if (!IsUBKLoaded())
+        {
+            debug_print(L"UBK not loaded, loading user-bound key");
+
+            // Convert to the expected DeveloperTypes struct
+            veil::vtl1::developer_types::keyCredentialCacheConfig convertedConfig;
+            convertedConfig.cacheOption = keyCredentialCacheConfiguration.cacheOption;
+            convertedConfig.cacheTimeoutInSeconds = keyCredentialCacheConfiguration.cacheTimeoutInSeconds;
+            convertedConfig.cacheUsageCount = keyCredentialCacheConfiguration.cacheUsageCount;
+
+            debug_print(L"Created mutableKeyCredentialCacheConfig");
+
+            // Load the user-bound key from secured encryption key bytes
+            // Note: securedEncryptionKeyBytes contains the user-bound key data, not raw key material
+            auto loadedKeyBytes = veil::vtl1::userboundkey::enclave_load_user_bound_key(
+                helloKeyName,
+                convertedConfig,
+                pinMessage,
+                windowId,
+                const_cast<std::vector<std::uint8_t>&>(securedEncryptionKeyBytes));
```

Comment:
>const_cast<std::vector<std::uint8_t>&> [](http://example.com/codeflow?start=16&length=38)

correctness: don't require const_cast in the enclave_load_user_bound_key api surface